packages/sketch/src/commands/pictograms/shared.js

```diff
@@ -0,0 +1,220 @@
+/**
```

Comment:
There were some updates to this logic on the icon side to support using the raw asset from the package that we might want to pull over: https://github.com/carbon-design-system/carbon/blob/master/packages/sketch/src/commands/icons/shared.js

Some of the notable changes:

- No longer requires `build-info.json`
- No longer requires `metadata.json` from the package and instead uses a [`generated` folder with metadata](https://github.com/carbon-design-system/carbon/blob/master/packages/sketch/tasks/build.js) for icons and pictograms that uses the asset plugin
- Consolidate logic for building the artboards into a common helper function: https://github.com/carbon-design-system/carbon/blob/master/packages/sketch/src/commands/icons/shared.js#L77

I was trying to structure it for this work but could easily not relate 👀 Let me know! I think the only hard requirement is using the asset data instead of the optimized data for rendering an icon